torch/ao/sparsity/experimental/data_sparsifier/lightning/tests/test_callbacks.py

```diff
@@ -78,5 +105,121 @@ def test_post_training_callback(self):
         self.run_all_checks(pl_module, callback, sparsifier_args)
 
 
+class _TrainingAwareCallbackTestCase(TestCase):
+    """Class to test in-training version of lightning callback
+    Simulates model training and makes sure that each hook is doing what is expected
+    """
+    def _check_on_train_start(self, pl_module, callback, sparsifier_args, scheduler_args):
+
+        callback.on_train_start(42, pl_module)  # 42 is a dummy value
```

Comment:
42 --> should we provide a DummyTrainer to match the expected type?

Reply:
Adding DummyTrainer seemed to raise some backend cuda error. Since, we are testing each function, it should be okay for now I think.